app/interactors/find_or_create_organization_model.rb

```diff
@@ -0,0 +1,34 @@
+class FindOrCreateOrganizationModel < ApplicationInteractor
+  def call
+    context.organization = find_or_create_organization
+
+    return if context.organization.valid?
+
+    context.fail!(error: :invalid_organization_params)
+  end
+
+  def rollback
+    context.organization.destroy if can_destroy?
+  end
+
+  private
+
+  def find_or_create_organization
+    find_organization ||
+      create_organization
+  end
+
+  def find_organization
+    context.organization = Organization.where(context.organization_params).first
```

Comment:
nitpick: autant laisser la db faire le boulal

```suggestion
    context.organization = Organization.find_by(context.organization_params)
```

Reply:
indeed